### PR TITLE
Create runtime bootstrap page

### PR DESCRIPTION
**Pull request by bulislaw**
New runtime bootstrap page documenting user hooks and retargeting.

@AnotherButler @sg-

### CODE REVIEW

File: docs/reference/runtime/Bootstrap.md [hunk 1]
```diff
@@ -1,12 +1,18 @@
 ## Bootstrap
 
-From a reset state, you can expect the following hooks and conditions. You can find more information in [`mbed_boot.c`](https://github.com/ARMmbed/mbed-os/blob/7b428916f5b35da28a1b18e2c149c1115bb89d92/rtos/TARGET_CORTEX/mbed_boot.c).
```
Comment (sg-) [inline docs/reference/runtime/Bootstrap.md]: While the function calls are not ideal here the concept of RAM initialization, clock initialization, CTORs etc seems important.

File: docs/reference/runtime/Bootstrap.md [hunk 2]
```diff
@@ -1,12 +1,18 @@
...
+
+### Retargeting
+
+Mbed OS redefines multiple standard C library functions to enable them to work in a predictable and familiar way on a remote embedded target device:
+
+- `stdin`, `stdout`, `stderr` - These file descriptors are pointing to the serial interface to enable users to use standard input/output functions, such as `printf` or `getc`.
+- `fopen`, `fclose`, `fwrite`, `fread`, `fseek` and other standard file operations - Enable the user to work with the serial interface, as well as the built-in file system.
+- `opendir`, `readdir`, `closedir` and other standard directory operations - Enable users to use built in file system.
+- `exit` - It causes the board to stop current execution, flush the standard file handles, close the semihosting connection and enter an infinite loop. If the return code indicates an error, the board blinks error patterns on the built-in LED.
+- `clock` - Overloaded to use platform's microsecond ticker.
```
Comment (sg-) [inline docs/reference/runtime/Bootstrap.md]: I dont recall ever seeing this. I'm intrigued

Comment (bulislaw) [reply docs/reference/runtime/Bootstrap.md]: https://github.com/ARMmbed/mbed-os/blob/master/platform/mbed_retarget.cpp#L1050

File: docs/reference/runtime/Bootstrap.md (hunk 1 above)
Comment (bulislaw) [reply docs/reference/runtime/Bootstrap.md]: Is it important from user point of view? I thought to document it in the porting guide, but from user's point of view it's all done by when the execution reaches main or mbed_main

Comment (sg-) [reply docs/reference/runtime/Bootstrap.md]: The specifics of the calls are not important but the concept of how the system works is important.

**Comment by AnotherButler** [comment]
@sg- Are we waiting on an answer to the mbed_main question, or can this go in?

**Comment by bulislaw** [comment]
Sorry missed this one, lets merge it for now. I have some changes to mbed boot in works, if we decide to merge it then I'll submit changes to the docs.

**Pull request merged**